data/config-schema/src/main/resources/net/corda/schema/configuration/p2p.gateway/1.0/corda.p2p.gateway.json

```diff
@@ -5,24 +5,36 @@
   "description": "Configuration schema for the P2P gateway.",
   "type": "object",
   "properties": {
-    "hostAddress": {
-      "description": "The host name or IP address where the HTTP server will bind.",
-      "type": "string",
-      "format": "hostname",
-      "default": "0.0.0.0"
-    },
-    "hostPort": {
-      "description": "The HTTP port.",
-      "type": "integer",
-      "default": 8080,
-      "minimum": 0,
-      "maximum": 65535
-    },
-    "urlPath": {
-      "description": "The url path the gateway server will be listening to.",
-      "type": "string",
-      "default": "/",
-      "pattern": "^\\/[\\/.a-zA-Z0-9-]*$"
+    "servers": {
+      "type": "array",
+      "minItems": 1,
+      "description": "The HTTP servers. The pair hostAddress and hostPort must be unique for each server.",
+      "default": [{}],
+      "items": {
+        "type":"object",
+        "description": "An HTTP server details.",
+        "properties": {
+          "hostAddress": {
+            "description": "The host name or IP address where the HTTP server will bind.",
+            "type": "string",
+            "format": "hostname",
+            "default": "0.0.0.0"
+          },
+          "hostPort": {
+            "description": "The HTTP port.",
+            "type": "integer",
+            "default": 8080,
```

Comment:
Related to the comment above on defaulting. If we add the default object at the array level, should we remove the default at the type level for ports?

I am thinking of the scenario where someone provides multiple entries, then the default won't make a lot of sense since it will clash.